pkg/handler/user_email_changed.go

```
@@ -23,23 +24,36 @@ func NewUserEmailChanged(s pkg.UserStore) *UserEmailChanged {
 
 // Handle is the user email changed message handler.
 func (u *UserEmailChanged) Handle(ctx context.Context, m *message.Message) error {
-	defer m.Ack(true)
-
 	user := new(pkg.User)
 	if err := json.Unmarshal(m.Body, user); err != nil {
 		log.Printf("failed to unmarshal message body: %v", err)
+		if err := m.Ack(false); err != nil {
+			log.Printf("failed to ack message: %v", err)
+			return err
```

we can remove this line, let's return the first error from `json.Unmarshal`